### PR TITLE
Clean TypeMeta's and ListMeta's description and force pruning 

### PR DESCRIPTION
**Pull request by soltysh**
/assign @sttts

### CODE REVIEW

File: config/v1/types.go [hunk 1]
```diff
@@ -167,6 +167,7 @@ type AdmissionPluginConfig struct {
 	// Configuration is an embedded configuration object to be used as the plugin's
 	// configuration. If present, it will be used instead of the path to the configuration file.
 	// +nullable
+	// +kubebuilder:validation:PreserveUnknownFields
 	Configuration runtime.RawExtension `json:"configuration"`
```
Comment (sttts) [inline config/v1/types.go]: this is an embedded resource I believe.

**Comment by soltysh** [comment]
Now that https://github.com/openshift/release/pull/5484 merged
/retest

File: hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk [hunk 2]
```diff
@@ -1,29 +1,22 @@
 self_dir :=$(dir $(lastword $(MAKEFILE_LIST)))
 
-CONTROLLER_GEN_VERSION ?=v0.2.1
-CONTROLLER_GEN_TEMP ?=$(PERMANENT_TMP_GOPATH)/src/sigs.k8s.io/controller-tools
-controller_gen_gopath =$(shell realpath -m $(CONTROLLER_GEN_TEMP)/../..)
-CONTROLLER_GEN ?=$(CONTROLLER_GEN_TEMP)/controller-gen
+CONTROLLER_GEN ?=$(PERMANENT_TMP_GOPATH)/bin/controller-gen
```
Comment (sttts) [inline hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk]: why not give it a unique name? This way, when switching branches, you easily end up with the wrong generator.

File: hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk [hunk 3]
```diff
@@ -1,29 +1,22 @@
...
 ensure-controller-gen:
 ifeq "" "$(wildcard $(CONTROLLER_GEN))"
-	$(info Installing controller-gen into "$(CONTROLLER_GEN)")
-	mkdir -p '$(CONTROLLER_GEN_TEMP)'
-	git clone -b '$(CONTROLLER_GEN_VERSION)' --single-branch --depth=1 https://github.com/sttts/controller-tools.git '$(CONTROLLER_GEN_TEMP)'
-	@echo '$(CONTROLLER_GEN_TEMP)/../..'
-	cd '$(CONTROLLER_GEN_TEMP)' && export GO111MODULE=on GOPATH='$(controller_gen_gopath)' && $(GO) mod vendor 2>/dev/null && $(GO) build -mod=vendor ./cmd/controller-gen
+	$(info Installing controller-gen into '$(CONTROLLER_GEN)')
+	mkdir -p '$(controller_gen_dir)'
+	curl -s -f -L https://github.com/openshift/kubernetes-sigs-controller-tools/releases/download/v0.2.1-37-ga3cca5d/controller-gen-$(GOHOSTOS)-$(GOHOSTARCH) -o '$(CONTROLLER_GEN)'
```
Comment (sttts) [inline hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk]: `s/v0.2.1-37-ga3cca5d/$(CONTROLLER_GEN_VERSION)/`

File: hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk (hunk 2 above)
Comment (sttts) [reply hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk]: `controller-gen-$(CONTROLLER_GEN_VERSION)`

File: hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk [hunk 4]
```diff
@@ -1,29 +1,22 @@
 self_dir :=$(dir $(lastword $(MAKEFILE_LIST)))
 
-CONTROLLER_GEN_VERSION ?=v0.2.1
```
Comment (sttts) [inline hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk]: CONTROLLER_GEN_VERSION ?=v0.2.1-37-ga3cca5d

File: hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk (hunk 2 above)
Comment (soltysh) [reply hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk]: Honestly, when switching branches `make clean` is needed, and `make update` in this repo is not that lengthy so I don't think that makes it worth the effort.

Comment (sttts) [reply hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk]: Remember Tim Hockin's twitter comment about make semantics? Having to run make clean is such a violation.

**Comment by wking** [comment]
Context around this change [here][1] and [here][2] (metadata schema/values are centrally enforced, although I'm not sure what that means for `oc explain ...`, etc.

[1]: https://github.com/openshift/cluster-version-operator/pull/266#discussion_r339727611
[2]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190425-structural-openapi.md#metadata

**Pull request merged**